sporkfish/searcher/move_ordering.py

```diff
@@ -47,13 +47,10 @@ def evaluate(self, board: Board, move: chess.Move) -> float:
             int: The heuristic value of the capturing move based on the value of the captured piece.
         """
 
-        captured_piece = board.piece_at(move.to_square)
-        moving_piece = board.piece_at(move.from_square)
-
         if (
-            captured_piece is not None
-            and moving_piece is not None
-            and board.is_capture(move)
+            board.is_capture(move)
+            and (captured_piece := board.piece_at(move.to_square))
+            and (moving_piece := board.piece_at(move.from_square))
```

Comment:
Interesting syntax, I never seen it before - TIL

https://stackoverflow.com/questions/26000198/what-does-colon-equal-in-python-mean

Reply:
Fascinating, this actually made a significant perf improvement. Nice catch